src/lib/Components/Inbox/Conversations/Preview/InvoicePreview.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from "react"
+import { Image, Text, TouchableHighlight } from "react-native"
+import * as Relay from "react-relay"
+import styled from "styled-components/native"
+
+import colors from "../../../../../data/colors"
+import fonts from "../../../../../data/fonts"
+
+import SerifText from "../../../Text/Serif"
+
+import InvertedButton from "../../../Buttons/InvertedButton"
+
+import AttachmentPreview, { AttachmentProps } from "../Preview/Attachment/AttachmentPreview"
+
+const Container = styled.View`
+  border-width: 1;
+  border-color: ${colors["gray-regular"]};
+  flex: 1;
+  flex-direction: row;
+`
+
+const TextContainer = styled.View`
+  flex: 1;
+  flex-direction: column;
+  margin-left: 25;
+  align-self: center;
+`
+
+const Icon = styled(Image)`
+  resize-mode: contain;
+  width: 40;
+  margin-top: 12;
+  margin-left: 12;
+  margin-bottom: 12;
+`
+
+const CanceledLabel = styled.Text`
+  color: ${colors["gray-medium"]};
+  font-weight: bold;
+`
+const RefundedLabel = styled.Text`
+  color: ${colors["gray-medium"]};
+  font-weight: bold;
+`
+const PaidLabel = styled.Text`color: ${colors["green-regular"]};`
+const PayButtonContainer = styled.View`
+  height: 25;
+  width: 71;
+`
+
+const PaymentRequest = styled.Text`
+  font-family: ${fonts["garamond-regular"]};
+  font-size: 14;
+`
+
+interface Props extends AttachmentProps, RelayProps {}
+
+interface InvoiceStateButtonProps {
+  state: string | null
+}
+
+const InvoiceStateButton: React.SFC<InvoiceStateButtonProps> = ({ state }) => {
+  if (state === "PAID") {
+    return <PaidLabel>PAID</PaidLabel>
+  }
+  if (state === "VOID") {
+    return <CanceledLabel>CANCELED</CanceledLabel>
+  }
+  if (state === "REFUNDED") {
+    return <RefundedLabel>REFUNDED</RefundedLabel>
+  }
+  if (state === "UNPAID") {
+    return (
+      <PayButtonContainer>
+        <InvertedButton text="PAY" />
+      </PayButtonContainer>
+    )
+  }
+}
+
+export const InvoicePreview: React.SFC<Props> = ({ invoice, onSelected }) =>
+  <AttachmentPreview attachment={invoice as any} onSelected={onSelected}>
```

Comment:
The `AttachmentPreview` component is only meant for items in the `attachments` list, an invoice is not such an item. I think the only thing you’re re-using this component for is the touch gesture? In which case you should just use a `TouchableHighlight` component like the `AttachmentPreview` component does internally.

Currently this misuse leads to the following warning (with Relay Modern, not sure if you got to see an error with classic):

> ExceptionsManager.js:71 Warning: RelaySelector: Expected object to contain data for fragment `AttachmentPreview_attachment`, got `{"__dataID__":"3","payment_url":"http://invoicing-demo-partner.lewitt-web-public-staging.artsy.net/invoices/371/EbaJoqyRKauLP6qYv86kyCos","state":"UNPAID","total":"$45,678"}`. Make sure that the parent operation/fragment included fragment `...AttachmentPreview_attachment`.

Reply:
👍 fixed in https://github.com/artsy/emission/pull/728, thx for pointing out! I missed this warning...